src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/DynamicObjectResolverOrderTest.cs

```diff
@@ -135,5 +154,45 @@ public ClassWithMissingKeyPositions()
             [Key(3)]
             internal string Memo { get; set; }
         }
+
+        [MessagePackObject(true)]
+        internal class ChildWithKeyAsPropertyName : ParentWithKeyAsPropertyName
+        {
+            [SerializationConstructor]
+            private ChildWithKeyAsPropertyName()
+                : base()
+            {
+            }
+
+            public ChildWithKeyAsPropertyName(string parentProp, long childProp)
+                : base(childProp)
+            {
+                ParentProp = parentProp;
+            }
+
+            public string ParentProp { get; private set; }
+        }
+
+        [MessagePackObject(true)]
+        internal class ParentWithKeyAsPropertyName
+        {
+            [SerializationConstructor]
+            private protected ParentWithKeyAsPropertyName()
+            {
+            }
+
+            public ParentWithKeyAsPropertyName(long childProp)
+            {
+                ChildProp = childProp;
+            }
+
+            [Key(0)]
```

Comment:
Why are you setting `Key(0)` here on a type where the property name is meant to be the key? I'm not sure what the expected behavior is in this case.